File: src/ros2_bridge/mappers/geometry_msgs/point_stamped.rs
//! Typed mapper for `geometry_msgs/msg/PointStamped`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn point_stamped_to_bus(msg: ros_env::geometry_msgs::msg::PointStamped) -> crate::geometry_msgs::msg::v1::PointStamped {
    crate::geometry_msgs::msg::v1::PointStamped {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        point: Some(crate::ros2_bridge::mappers::geometry_msgs::point::point_to_bus(msg.point)),
    }
}

pub(crate) fn point_stamped_to_ros(bus: crate::geometry_msgs::msg::v1::PointStamped) -> ros_env::geometry_msgs::msg::PointStamped {
    ros_env::geometry_msgs::msg::PointStamped {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(bus.header.unwrap_or_default()),
        point: crate::ros2_bridge::mappers::geometry_msgs::point::point_to_ros(bus.point.unwrap_or_default()),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsPointStampedMapper;

impl TypedTopicMapper for GeometryMsgsPointStampedMapper {
    type Ros = ros_env::geometry_msgs::msg::PointStamped;
    type Bus = crate::geometry_msgs::msg::v1::PointStamped;

    fn type_name(&self) -> &'static str {
        "geometry_msgs/msg/PointStamped"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(point_stamped_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(point_stamped_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/point_stamped.py
"""Generated mapper for `geometry_msgs/msg/PointStamped`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros
from robot_bus.ros2_bridge.mappers.geometry_msgs.point import point_to_bus, point_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.geometry_msgs.msg.v1 import PointStamped as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def point_stamped_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.point.CopyFrom(point_to_bus(msg.point))
    return bus


def point_stamped_to_ros(bus):
    from geometry_msgs.msg import PointStamped as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.point = point_to_ros(bus.point)
    return out


class GeometryMsgsPointStampedMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from geometry_msgs.msg import PointStamped as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return point_stamped_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return point_stamped_to_ros(bus)
